### ISSUE TITLE
IOS Production | Developer Error | Invalid Product ID

### ISSUE THREAD

**Issue by sabinayakc** [dooboolab/react-native-iap#2285]
> Please use the [Discussion board](https://github.com/dooboolab/react-native-iap/discussions) if you want to get some help. Please use issues to report bugs.

**Description**

- IOS Subscription works on simulator and on device for debug builds.
- Submitted my app to review with In app purchase for the first time
- All Agreeements and tax information are valid
- In-App Purchase got rejected but App was approved for release.
- In App Purchase screen is now blank with the error
`E_DEVELOPER_ERROR | Invalid product ID.`

**Expected Behavior**

- Should be able to get subscriptions information.

**Actual Behavior**
- In App Purchase screen is now blank with the error
`E_DEVELOPER_ERROR | Invalid product ID.`

**Environment:**

- react-native-iap: "^12.4.9",
- react-native: 0.70.2
- Platforms: IOS

[Optional] **Additional Context**

- Since the app was approved but the in app purchase rejected, do I have to re-submit a new build?

**Comment by andresesfm** [comment]
The error indicates a likely missconfiguration in the app store skus

**Comment by sabinayakc** [comment]
@acarkaan . 

My issue got finally resolved. 

If you are submitting your first IAP and along with a new app version, your app might get approved but your IAP can be rejected.

This will cause the error `E_DEVELOPER_ERROR | Invalid product ID` in your production app since it cannot retrieve the subscriptions which are not approved in App Store Connect.

To fix this,
- Go to your Subscription or In-App Purchase and check for details.
- In my case it was subscription which were rejected so I had to:
   -  Upload Screenshots for each subscription.
   -  Update the Localization Subscription Title *(Localization is reviewed separately so make sure you have an acceptable title such as `1 Month Premium` or `Monthly Premium etc.`.)*
   - After saving my changes, I re-submitted and sent it to review
   - The status of the subscription changed to `Waiting for Review`   
   > Note: You don't need to submit a new build if your app was approved. Subscription approval takes around 48 to 72 hours after making changes.

Hope this helps.
